src/test/java/org/jabref/gui/exporter/ExportToClipboardActionTest.java

```diff
@@ -0,0 +1,113 @@
+package org.jabref.gui.exporter;
+
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.gui.ClipBoardManager;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.Globals;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.LibraryTab;
+import org.jabref.gui.util.CurrentThreadTaskExecutor;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.logic.exporter.Exporter;
+import org.jabref.logic.exporter.ExporterFactory;
+import org.jabref.logic.exporter.SavePreferences;
+import org.jabref.logic.exporter.TemplateExporter;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.layout.LayoutFormatterPreferences;
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.logic.xmp.XmpPreferences;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.model.metadata.MetaData;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyCollection;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ExportToClipboardActionTest {
+
+    private ExportToClipboardAction exportToClipboardAction;
+    private LibraryTab libraryTab = mock(LibraryTab.class);
+    private JabRefFrame jabRefFrame = mock(JabRefFrame.class);
+    private DialogService dialogService = spy(DialogService.class);
+    private ExporterFactory exporterFactory;
+    private ClipBoardManager clipBoardManager = mock(ClipBoardManager.class);
+    private TaskExecutor taskExecutor;
+    private List<BibEntry> selectedEntries;
+    private BibDatabaseContext databaseContext = mock(BibDatabaseContext.class);
+
+    @BeforeEach
+    public void setUp() {
+//        if (Globals.prefs == null) {
```

Comment:
@Siedlerchr When I run this test, I get the error:
"Cannot invoke "org.jabref.preferences.JabRefPreferences.getFilePreferences()" because "org.jabref.gui.Globals.prefs" is null"
This is because line 84 in ExportToClipboardAction, it is calling:
.filter(exporter -> exporter.getName().equals(**Globals.prefs.getImportExportPreferences().getLastExportExtension()**))

I know we should not depend on Globals or alter the value of Globals.prefs in unit tests, but I was just trying if making Globals.prefs not null would solve the problem and it did. In this situation, how should we get arround this error the right way?

Thanks,
Ning

Reply:
Hi, as you cannot mock the Globals away, the only solution is to refactor the ExportToClipBoardAction, extend the constructor with an additional argument for PreferenceServices. As this is a bit more complicated, I just did this to show you how to do this if you encouter this again.

Reply:
Alternatively, you can merge this branch https://github.com/JabRef/jabref/tree/ningxie1991-a3-nx into yours

Reply:
@Siedlerchr Thanks for the quick reply and help! I have merged your change into my branch. I will fix the test case now.

Reply:
@Siedlerchr For some reason this pull requests suddenly includes 19 changed files. I think something happened when I merge your branch to mine. I replaced this pull request with a new one #7636. I am closing this pull request. Please do not merge this. Thanks.